### ISSUE TITLE
InvalidAbstractName should not throw error on *TestCase class

### ISSUE THREAD

**Issue by Soullivaneuh** [djoos/Symfony-coding-standard#121]
Example:

```
FILE: /code/tests/Security/Authorization/Voter/VoterTestCase.php
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 16 | ERROR | Abstract class name is not prefixed with "Abstract"
----------------------------------------------------------------------
```

But it's not true. See Symfony code source. `*TestCase` classes are not prefixed by `Abstract`.

Workaround:

```xml
<rule ref="Symfony.NamingConventions.ValidClassName.InvalidAbstractName">
    <exclude-pattern>*TestCase.php</exclude-pattern>
</rule>
```

**Comment by wickedOne** [comment]
as stated [over here](https://github.com/djoos/Symfony-coding-standard/issues/101#issuecomment-366518301) as well, personally i don't think referencing to symfony's test case classes or code in general is a valid reason for an adapted version of what is stated in their coding standard.

you can easily exclude your test case files if you don't agree with the coding standard if it comes to unit tests. disabling the rule by default would "break" this coding standard to those who actually want to apply the same rules to both their regular code and their test cases...

**Comment by hkdobrev** [comment]
From Symfony coding standard:

> Prefix all abstract classes with `Abstract` except PHPUnit `*TestCase`. Please note some early Symfony classes do not follow this convention and have not been renamed for backward compatibility reasons. However all new abstract classes must follow this naming convention;
> 
> https://symfony.com/doc/current/contributing/code/standards.html

I'd like to enforce that rule in our own code with this standard, but it requires abstract test cases to have the abstract prefix as well.

**Comment by Soullivaneuh** [comment]
> disabling the rule by default would "break"

I'm not talking about disabling the rule but make an exception on `TestCase` classes.

**Comment by djoos** [comment]
Thanks for this thread - IMO the coding standard shouldn't provide the exclusion, which goes against what the standard sets out (ie. an abstract class name needs to be not prefixed with "Abstract"). With that in mind, I'd push that back to your CI setup to have `TestCase` classes checked differently.

Hope this helps!
